### ISSUE TITLE
Typing Inconsistency for "v1.SearchSLOResponse" on GET /api/v1/slo/search

### ISSUE THREAD

**Issue by Vroomfrondal** [DataDog/datadog-api-client-typescript#1461]
**Describe the bug**
Bad key typing on the `v1.SearchSLOResponse` typescript type for GET /api/v1/slo/search. Proper response type is in `snake_case` while intellisense suggests `camelCase` for the property type.

Example 1: 
```ts
const type = sloData.data.attributes.slos[0].data.attributes?.sloType // actual property is `.slo_type`
``` 

Example 2: 
```ts
const status = sloData.data.attributes.slos[0].data.attributes?.overallStatus // actual property is `.overall_status`
``` 

**To Reproduce**
1) in a typescript environment create a GET request to `https://api.datadoghq.com/api/v1/slo/search`
2) Ensure the response type is using `v1.SearchSLOResponse`
3) View the response data and observe examples listed above within code-editors intellisense 

**Expected behavior**
- When using the property in code you should see a `snake_case` suggestion. 

**Screenshots**
![Screenshot 2023-12-19 at 3 01 46 PM](https://github.com/DataDog/datadog-api-client-typescript/assets/49052244/b4a8e960-17cb-44aa-9125-e579c772f8d3)

**Environment and Versions (please complete the following information):**
- datadog/datadog-api-client version 1.19.0
- Typescript

**Comment by therve** [comment]
Hi,

Do you have a reproducible example? Models are creating camelCase properties, which are available in my testing. Here is what I run:

```
import { client, v1 } from "@datadog/datadog-api-client";

const configuration = client.createConfiguration();
const apiInstance = new v1.ServiceLevelObjectivesApi(configuration);

const params: v1.ServiceLevelObjectivesApiSearchSLORequest = {
  pageSize: 20,
  pageNumber: 0,
};

apiInstance
  .searchSLO(params)
  .then((data: v1.SearchSLOResponse) => {
    console.log(data.data.attributes.slos[0].data.attributes?.sloType)
    console.log(data.data.attributes.slos[0].data.attributes?.overallStatus)
  })
  .catch((error: any) => console.error(error));
```

Thanks.

**Comment by Vroomfrondal** [comment]
Hey @therve 

So that example you provided works for you? Because if asked that's the example I would provide. For me `sloType` and `overallStatus` are actually `slo_type` and `overall_status`.  Requiring a @ts-expect-error for intellisense to ignore. 

There are similar (but unrelated) issues like this throughout. For example, on the POST endpoint to create embedded graphs - the "size" parameter is an enum of strings (by fact and documentation) but vscode is saying it should be a `number` type.

**Comment by therve** [comment]
Yes that example work for me. What's your typescript version? The embedded graphs doesn't generate clients. so it's a different issue indeed.

**Comment by Vroomfrondal** [comment]
5.0.4, Admittedly a few behind. Maybe this was solved already?
